### ISSUE TITLE
Multi channel 32F Mat setTo NaN is not setting to NaN

### ISSUE THREAD

**Issue by gmedan** [opencv/opencv#10507]
##### System information (version)
- OpenCV => 3.3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017 (15.5.2)

##### Detailed description

Setting 3 or 4 channel 32F Mat to NaN using mask actually sets the value to 3.40282347e+38.
1 or 2 channels works fine.
64F works fine.
No mask works fine.
Problem is not reproduced in opencv 3.2.0

##### Steps to reproduce
```.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main(void)
{
    for (int chans = 1; chans <= 4; chans++)
    {
        cv::Mat A(1, 1, CV_MAKE_TYPE(CV_32F, chans), cv::Scalar::all(0));
        A.setTo(cv::Scalar::all(std::numeric_limits<float>::quiet_NaN()), cv::Mat(1, 1, CV_8UC1, cv::Scalar::all(255)));
        std::cout << "Number of NaN pixels in " << chans << " channels image is " << (cv::countNonZero((A == A) == 0)) << " (should be 1)" << std::endl;
    }
}
```

output:
Number of NaN pixels in 1 channels image is 1 (should be 1)
Number of NaN pixels in 2 channels image is 1 (should be 1)
Number of NaN pixels in 3 channels image is 0 (should be 1)
Number of NaN pixels in 4 channels image is 0 (should be 1)

**Comment by alalek** [comment]
> 3.3.1

Please try the latest code from 3.4 branch.

**Comment by take1014** [comment]
I ran this code with the latest 3.4,  but all channels have NaN values.
Is this bug fixed already? Should I close this pull request?

**Comment by alalek** [comment]
Looks like it is get fixed here: #11192 (resolves similar problem with `Inf` values)

**Comment by take1014** [comment]
@alalek 
Thank you for a comment.
In #11192 , only the test with INF values has been done, but is it unnecessary to add a test with NAN values?
